### ISSUE TITLE
Private Key import fails with OpenSSL 3.0

### ISSUE THREAD

**Issue by debreczeni** [GemHQ/money-tree#63]
Error message when trying to import a private key using OpenSSL 3.0.5:

```
MoneyTree::Master.from_bip32(private)
```

results in this exception:

```
Exception: OpenSSL::PKey::PKeyError: pkeys are immutable on OpenSSL 3.0
--
 0: /Users/username/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/bundler/gems/money-tree-5d80ffb7302b/lib/money-tree/key.rb:58:in `private_key='
 1: /Users/username/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/bundler/gems/money-tree-5d80ffb7302b/lib/money-tree/key.rb:58:in `import'
 2: /Users/username/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/bundler/gems/money-tree-5d80ffb7302b/lib/money-tree/key.rb:46:in `initialize'
 3: /Users/username/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/bundler/gems/money-tree-5d80ffb7302b/lib/money-tree/node.rb:36:in `new'
 4: /Users/username/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/bundler/gems/money-tree-5d80ffb7302b/lib/money-tree/node.rb:36:in `parse_out_key'
 5: /Users/username/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/bundler/gems/money-tree-5d80ffb7302b/lib/money-tree/node.rb:31:in `from_bip32'
```

**Comment by q9f** [comment]
Confirmed on CI with Ruby 3.1 and Ubuntu 22.4, interestingly, it works with Ruby 3.0 still.

**Comment by q9f** [comment]
fixed in #64 #71
